### PR TITLE
Client - Allow specifying pipeline description when uploading

### PR DESCRIPTION
**Pull request by Ark-kun**
Fixes https://github.com/kubeflow/pipelines/issues/3825

### CODE REVIEW

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -672,7 +672,7 @@ def _get_workflow_json(self, run_id):
     workflow_json = json.loads(workflow)
     return workflow_json
 
-  def upload_pipeline(self, pipeline_package_path, pipeline_name=None):
+  def upload_pipeline(self, pipeline_package_path, pipeline_name=None, description: str = None):
```
Comment (numerology) [inline sdk/python/kfp/_client.py:675]: nit: can we unify the code style by wrapping '=' with two spaces for `pipeline_name` as well? Same below.

File: sdk/python/kfp/_client.py [hunk 2]
```diff
@@ -672,7 +672,7 @@ def _get_workflow_json(self, run_id):
...
     """Uploads the pipeline to the Kubeflow Pipelines cluster.
     Args:
       pipeline_package_path: Local path to the pipeline package.
```
Comment (numerology) [inline sdk/python/kfp/_client.py:678]: Let's also add description in the docstring.

**Comment by numerology** [comment]
/lgtm

Thanks!

File: sdk/python/kfp/_client.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/kfp/_client.py]: Done.

File: sdk/python/kfp/_client.py (hunk 2 above)
Comment (Ark-kun) [reply sdk/python/kfp/_client.py:683]: Done.

**Pull request merged**